### PR TITLE
BUG: fix mean for float 16 non-array inputs

### PR DESCRIPTION
**Pull request by juliantaylor**

### CODE REVIEW

File: numpy/core/tests/test_multiarray.py [hunk 1]
```diff
@@ -4380,6 +4380,12 @@ def setUp(self):
         self.omat = np.array([Decimal(repr(r)) for r in self.rmat.flat])
         self.omat = self.omat.reshape(4, 5)
 
+    def test_python_type(self):
+        for x in (np.float16(1.), 1, 1., 1+0j):
```
Comment (f0k) [inline numpy/core/tests/test_multiarray.py]: To have a test that triggers the bug fixed in this PR, you may want to add `[np.float16(1.)]` to this list (i.e., a float16 wrapped in a list). The others don't run into the `is_float16_result` and `not hasattr(a, "dtype")` case.

**Review by f0k — CHANGES_REQUESTED** [review]
That was fast! I'd suggest a minimal addition to the test.

File: numpy/core/tests/test_multiarray.py (hunk 1 above)
Comment (juliantaylor) [inline numpy/core/tests/test_multiarray.py]: all tests are scalars wrapped into a list, though just scalars might be useful too

Comment (f0k) [inline numpy/core/tests/test_multiarray.py]: > all tests are scalars wrapped into a list

Ooops, right. Great then! Good to merge from my perspective. Thank you for taking this!

**Review by f0k — APPROVED** [review]
Seems I cannot edit my previous review from "Request changes" to "Approve". Not that it matters too much, since I'm just a contributor without write access. Anyway, this is my new review which approves the PR :)

**Comment by charris** [comment]
Thanks Julian. Should I do the backport?

**Pull request merged**